File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/Repository/MagazineRepository.java
package abbruzzese.webpage.webpage.server.webpage.Repository;

import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import abbruzzese.webpage.webpage.server.webpage.entities.Magazine;
import java.util.List;
import java.util.Optional;

public interface MagazineRepository extends CrudRepository<Magazine,String>{

    Optional<Magazine> findByNombre(String nombre);

    @Query("SELECT m FROM Magazine m WHERE m.Año BETWEEN :startYear AND :endYear")
    List<Magazine> findRevistasByAnioRange(@Param("startYear") String startYear, @Param("endYear") String endYear);
    
      // // Método para obtener la última revista según fecha de creación (si la tienes) o alguna otra lógica
      @Query("SELECT m FROM Magazine m ORDER BY m.fecha DESC ")
      Optional<Magazine> findLastMagazine();


}

File: PHP_API/REPOSITORY/MagazineRepository.php
<?php 

namespace App\Repository;

use Doctrine\ORM\EntityRepository;
use App\Entity\Magazine;

class MagazineRepository extends EntityRepository{

    public function findByNombre($nombre):?Magazine
    {
        return $this->findOneBy(['nombre' => $nombre]);
    }


    public function findRevistasByAnioRange(string $startYear, string $endYear): array
    {
        $dql = "SELECT m FROM App\Entity\Magazine m WHERE m.anio BETWEEN :startYear AND :endYear";
        
        return $this->getEntityManager()
                    ->createQuery($dql)
                    ->setParameter('startYear', $startYear)
                    ->setParameter('endYear', $endYear)
                    ->getResult();
    }

    public function findLastMagazine(): ?Magazine
{
    $dql = "SELECT m FROM App\Entity\Magazine m ORDER BY m.fecha DESC";
    
    return $this->getEntityManager()
                ->createQuery($dql)
                ->setMaxResults(1)
                ->getOneOrNullResult();
}

}

?>